arangod/GeneralServer/HttpCommTask.cpp

```diff
@@ -416,15 +416,13 @@ void HttpCommTask<T>::processRequest() {
   // C functions like strchr that except a C string as input
   _request->body().push_back('\0');
   _request->body().resetTo(_request->body().size() - 1);
-
+  _url = std::string((_request->databaseName().empty() ? "" : "/_db/" + _request->databaseName())) +
+    (Logger::logRequestParameters() ? _request->fullUrl() : _request->requestPath());
```

Comment:
This pessimizes the case in which we don't log requests (not logging requests is the default, by the way).
Instead of unconditionally storing the url in an instance variable `_url`, we could make the log messages call a new function named `url()`, which computes the url value only when needed, and then stores the computed value in the `_url` instance variable.
That way we would avoid computation of `_url` in case there is no logging, and still benefit if the `_url` value is logged in 2 places.
That idea will only work however if the `url()` function can access `_request` and `_request` is still valid while `url()` is called. I don't know if that is the case or not, but we could try it.